pandas/tests/util/test_assert_frame_equal.py

```diff
@@ -9,6 +9,11 @@ def by_blocks(request):
     return request.param
 
 
+@pytest.fixture(params=['aaa', 'zzz'])
```

Comment:
this should also include DataFrame & Series as strings;

can you call this obj_fixture? (and rename by_blocks -> by_blocks_fixture)

Reply:
Will do. Is it enough to just include DataFrame and Series as strings in the fixture? Or should that somehow be made to disable specifying `obj=` in the tests, to test that expected default value is honoured?

Reply:
I think just DataFrame and Series as strings is fine ('aaa' and 'zzz' are kind of arbitrary). You'd have to then replace `obj` in your function signatures with `obj_fixture` and subsequently change invocations of these assert functions within those tests to `obj=obj_fixture`

Reply:
Sounds good. This is straining my knowledge of pep8 though!
```
{obj}\\.index values are different \\(33\\.33333 %\\)
\\[left\\]:  Index\\(\\['a', 'b', 'c'\\], dtype='object'\\)
\\[right\\]: Index\\(\\['a', 'b', 'd'\\], dtype='object'\\)""".format(obj=obj_fixture)
```
I'll read up to figure out how to shorten that last line, but if you see this in the meantime, could you help me out?

Reply:
Yea you can just break after `format(`

Reply:
K, done and pushed. When I revise my PR, I've been amending, then force-push. This time I forgot to fetch and reapply to the current master though. Is there a preferred workflow for revising pull requests?

Reply:
I would do this before any change:

```sh
git fetch upstream
git merge upstream/master
# if any merge conflicts fix up and
git merge --continue
```

Then you can just make new commits without amend and push without force

Reply:
It will be squashed and rebased when it's merge to pandas-dev/pandas though, right? I'll have to force-push after that, but I guess your recipe will keep the history organized until the PR is finally accepted?

Reply:
We squash before merge